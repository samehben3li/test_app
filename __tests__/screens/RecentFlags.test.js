import React from "react";
import renderer from "react-test-renderer";
import RecentFlags from "../../screens/RecentFlags";
import { MockedProvider } from "@apollo/client/testing";
import AuthProvider from "../../context/authContext";

describe("RecentFlags Screen", () => {
  it("renders without crashing", () => {
    const rendered = renderer
      .create(
        <MockedProvider>
          <AuthProvider value={{}}>
            <RecentFlags route={{ name: "createFlag" }} />
          </AuthProvider>
        </MockedProvider>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
