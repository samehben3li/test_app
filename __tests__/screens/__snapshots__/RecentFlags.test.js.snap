// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentFlags Screen renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#fff",
          "borderBottomEndRadius": 40,
          "borderBottomStartRadius": 40,
          "elevation": 24,
          "flexDirection": "row",
          "height": 150,
          "justifyContent": "space-between",
          "paddingBottom": 30,
          "paddingHorizontal": 40,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 12,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 52,
            "padding": 0,
            "width": 120,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 40,
              "resizeMode": "contain",
              "width": "auto",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "medium",
              "fontSize": 11,
              "height": 13,
              "padding": 0,
              "position": "relative",
              "textAlign": "left",
              "width": "100%",
            }
          }
        >
          Tomato Trolley
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 30,
            "marginBottom": 14,
            "width": 40,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": "100%",
              "resizeMode": "contain",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#000",
        "borderBottomEndRadius": 40,
        "borderBottomStartRadius": 40,
        "flexDirection": "row",
        "height": 60,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": "30%",
              "opacity": 0.5,
              "resizeMode": "contain",
              "tintColor": "#FFB955",
              "width": "100%",
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
          },
          Object {
            "borderLeftColor": "#F3B25580",
            "borderLeftWidth": 2,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": "30%",
              "opacity": 0.5,
              "resizeMode": "contain",
              "tintColor": "#FFB955",
              "width": "100%",
            },
            false,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Loading...
    </Text>
  </View>
</View>
`;
